OIPA/direct_indexing/custom_fields/activity_dates.py

```diff
@@ -0,0 +1,26 @@
+def activity_dates(data):
+    """
+    Requested by FCDO. Single valued fields.
+    Activity dates is one of four types: planned or actual, start or end.
+    start planned (type 1), start actual (type 2),
+    end planned (type 3), end actual (type 4)
+
+    Relevant fields:
+    'activity-date.type' and 'activity-date.iso-date'
+
+    :param data: reference to the activity in the data
+    """
+    try:
+        if 'activity-date' in data.keys():
+            for date in data['activity-date']:
+                # This approach was tested to be the fastest with 10.000 runs
+                i = 1
+                for s in ['start', 'end']:
+                    for ss in ['planned', 'actual']:
+                        if 'type' in date.keys() and 'iso-date' in data.keys():
```

Comment:
maybe this can be changed into set operations

Reply:
Not yet seeing the usefulness of sets here as the strings are concatenated with a separator and the order is important!
But if you could elaborate on the suggestion, maybe we could work out a solution! Maybe on Tuesday?